src/features/authentication/sign-up/SignUp.dao.ts

```diff
@@ -7,18 +8,34 @@ export class SignUpDao {
   async userExists(member: Partial<Member>): Promise<boolean> {
     if (await this.findMemberByUsername(member.userName)) return true
     if (await this.findMemberById(member.id)) return true
+    if (await this.findMemberByEmail(member.email)) return true
     return false
   }
 
   async findMemberByUsername(userName: string): Promise<Member | null> {
+    if (!userName) return null
     return await this.memberRepository.findOneBy({ userName: userName })
   }
 
   async findMemberById(id: number): Promise<Member | null> {
+    if (!id) return null
     return await this.memberRepository.findOneBy({ id: id })
   }
 
+  async findMemberByEmail(email: string): Promise<Member | null> {
+    if (!email) return null
+    return await this.memberRepository.findOneBy({ email: email })
+  }
+
   async saveMemberInDB(member: Partial<Member>): Promise<Member> {
     return await this.memberRepository.save(member)
```

Comment:
So, we have @BeforeInsert annotation in` Member.ts` entity right. To trigger this annotation while inserting we need to change code in dao file.